### ISSUE TITLE
"primary" billing location determination is naive

### ISSUE THREAD

**Issue by surfacehazard** [openemr/openemr#5773]
https://github.com/openemr/openemr/blob/52773322af5a42b2622f0ce285433be6a1247323/src/Services/FacilityService.php#L125

Currently (v.7.x), the most _recently created_ billing-enabled facility is considered the "primary" billing location, which is misleading for calling code that is using it as a default facility.

The "primary" billing location should be assignable. However, I think the facility table (and associated GUI) would need to be enhanced by an additional column to achieve this, either a flag or "priority" column I imagine.

**Comment by sunsetsystems** [comment]
The original intent was there would be only one billing facility. Is there now a need for more than one?

**Comment by stephenwaite** [comment]
Pretty sure it was added so that you could have a billing entity that had a different address than a billing location. There's a column in `facility` for this, `primary_business_entity`.

**Comment by stephenwaite** [comment]
calling this a bug because the gui isn't behaving as it should in facility admin

**Comment by sunsetsystems** [comment]
For what it's worth -- if a billing facility becomes obsolete but needs to be preserved, you could uncheck the "billing location" flag for it. It was intended to provide the address that payments would be sent to, so would appear in statements and perhaps other correspondence. Perhaps the bug is that these things are not well documented.

**Comment by surfacehazard** [comment]
> The original intent was there would be only one billing facility. Is there now a need for more than one?

Yes. The billing facility determines CMS-1500 Box 25 and 33 (and corresponding 835) information for claims. Payors have different requirements for those fields. For example, the US Department of Labor requires their own Provider ID be used in box 33a instead of a NPI. Other payors differ as to whether a provider or group TIN be used in Box 25, or a type-1 or type-2 NPI in 33a. 

Multiple billing facilities are used to address this.

**Comment by stephenwaite** [comment]
Remembered another user of multiple billing facilities which came up during the 2015 certification, https://github.com/openemr/openemr/issues/5386. We have to support different tax ids within OpenEMR.

**Comment by surfacehazard** [comment]
I think the ideal solution is to have claim fields customizable (overridable?) per insurance company/payor, but I limited the scope of this issue to just a misleading function.

**Comment by stephenwaite** [comment]
there is the option to assign provider numbers as well that could be easier to implement if it's not addressing that

**Comment by stephenwaite** [comment]
it's probably best to replace calls to `getPrimaryBillingLocation()` with [getPrimaryBusinessEntity()](https://github.com/openemr/openemr/blob/52773322af5a42b2622f0ce285433be6a1247323/src/Services/FacilityService.php#L89) where the selected primary billing facility is desired instead of an array of all billing locations
